frappe/model/document.py

```diff
@@ -310,6 +310,9 @@ def insert(
 		self.run_post_save_methods()
 		self.flags.in_insert = False
 
+		# run hooks
+		run_last_instance_of_webhooks(frappe.flags.webhooks_to_execute or [])
```

Comment:
Umm, does this solve your problem? IIRC your problem was same doc with multiple instances triggering changes using on_change where only first change was published via webhook. 


Your logic is mostly right but you'll need to execute this function using `db.after_commit` and not immediately during doc.save, otherwise your problem still persists.